File: CA3CalculatorPython/calculator/calculator.py
import math

class Calculator(object):
    """Functions of a Calculator"""
    def __init__(self):
        self.__userinput1 = 0
        self.__userinput2 = 0
        self.__myFunctions = { '+':'Add',
                               '-':'Subtract',
                               '*':'Multiply',
                               '/':'Divide',
                               '^':'Exponent',
                               's':'Squared',
                               'r':'Square Root',
                               'c':'Cubed',
                               '%':'Percent to Decimal',
                               '!':'Factorial'}
        self.__oneValueFunction = {'s':self.square, 'r':self.squareRoot, 'c':self.cube, '%':self.percentToDecimal,'!':self.factorial}
        self.__twoValueFunction = {'+':self.add, '-':self.subtract, '*' : self.multiply, '/':self.divide, '^':self.exponent}
        self.__oneValueSymbols = ['s','r','c','%','!']
        self.__twoValueSymbols = ['+','-','*','/','^']
        self.userfunction = ''
        self.userinput = False
        
    @property
    def userinput1(self):
        return self.__userinput1
    
    @userinput1.setter
    def userinput1(self, value):
        try:
            self.__userinput1 = float(value)
        except:
            raise ValueError
    
    @property
    def userinput2(self):
        return self.__userinput2
        
    @userinput2.setter
    def userinput2(self, value):
        try:
            self.__userinput2 = float(value)
        except:
            raise ValueError
         
    @property
    def oneValueSymbols(self):
        return self.__oneValueSymbols
        
    @property
    def twoValueSymbols(self):
        return self.__twoValueSymbols
    
    
    def add(self, number1, number2):
        return number1 + number2
        #return math.add(number1, number2)
        
    def cube(self, number):
        return number ** 3
    
    def divide(self, number1, number2):
        try:
            return number1 / number2
        except ZeroDivisionError:
            return "Can't divide by zero!"
    
    def exponent(self, base, power):
        return base ** power
    
    def factorial(self, number):
        try:
            if(number <2):
                return 1
            else:
                result = 1
                for i in range(2,number+1):
                    result *= i
                return result
        except ValueError as factorialError:
            return "Not possible to calculate factorial of a negative number"
    
    def multiply(self, number1, number2):
        return number1 * number2
    
    def percentToDecimal(self, number):
        return float(number) / 100
    
    def printFunction(self):
        mf = self.__myFunctions
        for k, v in mf.items():
            print('| {0} : {1} |'.format(k, v), end = '' )
    
    def square(self, number):
        return self.exponent(number, 2)
    
    def squareRoot(self, number):
        try:
            return math.sqrt(number)
        except ValueError as sqrtError:
            return "No square root of a negative number"
        
    def subtract(self, number1, number2):
        return number1 - number2
        
    def selectFunction(self, function, value1, value2=0):
        result = 0
        if function in self.__oneValueSymbols:
            oneFunc = self.__oneValueFunction
            for key in oneFunc:
                if(key == function):
                    result = oneFunc[function](value1)
                    return result
        elif function in self.__twoValueSymbols:
            twoFunc = self.__twoValueFunction
            for key in twoFunc:
                if (key == function):
                    result = twoFunc[function](value1, value2)
                    return result
    def checkQuit(self, input):
        if (input == 'q' or input =='Q'):
            self.userinput = True
            return print("Goodbye!")
        

File: CA3CalculatorC/CA3CalculatorC/Calculator.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;


namespace CA3CalculatorC
{
	public class Calculator
	{
		//Variables
		private static Dictionary<string, string> myFunctions = new Dictionary<string, string>()
		{
			{ "+","Add" },
			{ "-","Subtract" },
			{ "*","Multiply" },
			{ "/","Divide" },
			{ "^","Exponent" },
			{ "s","Squared" },
			{ "r","Square Root" },
			{ "c","Cubed" },
			{ "%","Percent to Decimal" },
			{ "!","Factorial" }
		};

		//private static string[] oneValueSymbols = { "s", "r", "c", "%", "!" };
		private static string[] twoValueSymbols = { "+", "-", "*", "/", "^" };

		//Properties
		public string[] TwoValueSymbols
		{
			get
			{
				return twoValueSymbols;
			}
		}
		
		public Dictionary<string, string> MyFunctions
		{
			get
			{
				return myFunctions;
			}			
		}
		public string UserFunction { get; set; }
		
		//Constructor
		//use default

		//Methods
		public static decimal Add(decimal value1, decimal value2)
		{
			return value1 + value2;
		}

		public static decimal Cube(decimal value)
		{
			return value * value * value;			
		}

		public static decimal Divide(decimal value1, decimal value2)
		{
			return value1 / value2;
		}

		public static double Exponent(double baseNumber, double power)
		{
			return Math.Pow(baseNumber, power);
		}

		public static int Factorial(int value)
		{

			if (value < 0)
			{
				throw new ArithmeticException("Not possible to calculate factorial value of negative numbers.");
			}
			if (value == 1 || value == 0)
			{
				return 1;
			}
			else
			{
				return value * Factorial(value - 1);
			}
		}

		public static decimal Multiply(decimal value1, decimal value2)
		{
			return value1 * value2;
		}

		public static decimal PercentToDecimal(decimal value)
		{
			return value / 100;
		}

		public void PrintFunction()
		{
			foreach (KeyValuePair<string, string> pair in myFunctions)
			{
				Console.Write("| {0} : {1} |", pair.Key, pair.Value);
			}
			Console.WriteLine();
		}

		public static decimal Square(decimal value)
		{
			return value * value;
		}

		public static double SquareRoot(double value)
		{
			if (value <=0)
			{
				throw new ArithmeticException("Invalid input - try again");
			}
			else
			{
				return Math.Sqrt(value);
			}
		}

		public dynamic SelectedFunction(string function, string input1, string input2 = "0")
		{
			switch (function)
			{				
				case "+":
					{
						try
						{
							return Calculator.Add(decimal.Parse(input1), decimal.Parse(input2));
						}
						catch (Exception)
						{
							throw new ArithmeticException("Invalid input - try again!");
						}
					 }
				case "c":
					try
					{
						return Calculator.Cube(decimal.Parse(input1));
					}
					catch (Exception e)
					{
						return e.Message.ToString();
					}
				case "-":
					try
					{
						return Calculator.Subtract(decimal.Parse(input1), decimal.Parse(input2));
					}
					catch (Exception e)
					{
						return e.Message.ToString();
					}
				case "*":
					try
					{
						return Calculator.Multiply(decimal.Parse(input1),decimal.Parse(input2));
					}
					catch (Exception e)
					{
						return e.Message.ToString();
					}
				case "/":
					try
					{
						return Calculator.Divide(decimal.Parse(input1), decimal.Parse(input2));
					}
					catch (DivideByZeroException e)
					{
						return e.Message.ToString();
					}
					
				case "^":
					try
					{
						return Calculator.Exponent(double.Parse(input1), double.Parse(input2));
					}
					catch (Exception e)
					{
						return e.Message.ToString();
					}
				case "s":
					try
					{
						return Calculator.Square(decimal.Parse(input1));
					}
					catch (Exception e)
					{
						return e.Message.ToString();
					}
				case "r":
					try
					{
						return Calculator.SquareRoot(double.Parse(input1));
					}
					catch (ArithmeticException e)
					{
						return e.Message.ToString();
					}

				case "%":
					try
					{
						return Calculator.PercentToDecimal(decimal.Parse(input1));
					}
					catch (Exception e)
					{
						return e.Message.ToString();
					}
				case "!":
					try
					{
						return Calculator.Factorial(int.Parse(input1));
					}
					catch (ArithmeticException e)
					{
						return e.Message.ToString();
					}
					

				default:
					return "Invalid input - try again";
			}

		}

		public static decimal Subtract(decimal value1, decimal value2)
		{
			return value1 - value2;
		}


	}
}
